src/components/ReservationModal.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import styled from 'styled-components';
+import { WHITE, BLACK } from '../colors';
+import { Modal } from 'antd';
+
+interface ReservationModalProps {
+    readonly status: string;
```

Comment:
```suggestion
enum ReservationModalType {
  OPEN = 'OPEN'
}

interface ReservationModalProps {
    readonly status: ReservationModalType;
```

You can make an enum to restrict which types are allowed for status.